lib/behavior.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+/**
+ * Built-in behaviors.
+ * These provide hooks into modifying Mocha's behavior for different use cases.
+ */
+
+var utils = require('./utils');
+
+/**
+ * Default Mocha behavior
+ */
+exports.Default = {
+  Suite: {
+    /**
+     * Runs the suite; called by `Suite.create()`.  Calls its callback fn
+     * with the Suite as its context.
+     * @this {Suite}
+     */
+    run: function() {
+      // FUTURE: async suites
+      this.fn.apply(this);
+    }
+  },
+  Runnable: {
+    /**
+     * Determines whether or not we should provide a nodeback parameter and
+     * expect it to be called
+     * @this {Runnable}
+     * @returns {boolean} `true` if we should run the test fn with a callback
+     */
+    shouldRunWithCallback: function() {
```

Comment:
So, IMO, this is too specific.

What may be better is supporting a function which answer the question "what arguments should we pass to the execution function?", and another "what context should the execution function run with?"

Obviously we're going to need to document this specification...